### PR TITLE
Basic authorization and authentication mechanism

### PR DESCRIPTION
**Pull request by loxlouro**

### CODE REVIEW

File: src/main/kotlin/com/reappa/rieltorapp/RieltorAppApplication.kt [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+import com.reappa.rieltorapp.services.RoleService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.CommandLineRunner
 import org.springframework.boot.autoconfigure.SpringBootApplication
 import org.springframework.boot.runApplication
 
 @SpringBootApplication
-class RieltorAppApplication
+class RieltorAppApplication(
+//    @Autowired
```
Comment (HowDidUGetOutOfBounds) [inline src/main/kotlin/com/reappa/rieltorapp/RieltorAppApplication.kt]: do not leave ghost code

File: src/main/kotlin/com/reappa/rieltorapp/config/BeanContainer.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+) {
+    @Bean
+    fun authenticationProvider(): DaoAuthenticationProvider {
+        val daoAuthenticationProvider = DaoAuthenticationProvider()
+        daoAuthenticationProvider.setPasswordEncoder(passwordEncoder)
+        daoAuthenticationProvider.setUserDetailsService(accountService)
+        return daoAuthenticationProvider
+    }
+
+
```
Comment (HowDidUGetOutOfBounds) [inline src/main/kotlin/com/reappa/rieltorapp/config/BeanContainer.kt]: to much whitespaces

File: src/main/kotlin/com/reappa/rieltorapp/dtos/RegistrationDto.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.reappa.rieltorapp.dtos
+
+import lombok.AllArgsConstructor
+import lombok.Data
+
+//@Data
```
Comment (HowDidUGetOutOfBounds) [inline src/main/kotlin/com/reappa/rieltorapp/dtos/RegistrationDto.kt]: remove ghost code

File: src/main/kotlin/com/reappa/rieltorapp/models/Role.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.security.core.GrantedAuthority
+@Entity
+@Data
+@Getter
+@Setter
+@RequiredArgsConstructor
+data class Role(
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+//    @get:JvmName("getRoleId")
```
Comment (HowDidUGetOutOfBounds) [inline src/main/kotlin/com/reappa/rieltorapp/models/Role.kt]: remove ghost code

File: src/main/kotlin/com/reappa/rieltorapp/services/AccountService.kt [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.stereotype.Service
+
+@Service
+@RequiredArgsConstructor
+//@EnableJpaRepositories
+class AccountService (
+    @Autowired
+    val accountRepository: AccountRepository,
+    ):UserDetailsService{
+//        @Bean
```
Comment (HowDidUGetOutOfBounds) [inline src/main/kotlin/com/reappa/rieltorapp/services/AccountService.kt]: remove ghost code

**Review by HowDidUGetOutOfBounds — CHANGES_REQUESTED** [review]

**Review by loxlouro — COMMENTED** [review]
Fixes after review

**Pull request merged**